--- kvazaar-0.7.0/src/Makefile.orig	2015-10-14 21:32:29.905680671 +0200
+++ kvazaar-0.7.0/src/Makefile	2015-10-14 21:32:59.209013433 +0200
@@ -123,7 +123,11 @@
   ifeq ($(TARGET_CPU_BITS), 64)
     ASFLAGS += -f elf64
   else
+    ifeq ($(TARGET_CPU_BITS), x32)
+      ASFLAGS += -f elfx32
+    else
     ASFLAGS += -f elf32
+    endif
   endif
   LIBKVAZAAR_SHARED = $(LIB)
   INSTALL_TARGETS += install-lib install-static
@@ -137,7 +137,11 @@
 ifeq ($(TARGET_CPU_ARCH) $(TARGET_CPU_BITS), x86 64)
   ASFLAGS += -DARCH_X86_64=1
 else
+  ifeq ($(TARGET_CPU_ARCH) $(TARGET_CPU_BITS), x86 x32)
+    ASFLAGS += -DARCH_X86_64=1
+  else
   ASFLAGS += -DARCH_X86_64=0
+  endif
 endif
 
 # Disable ASM optimizations if YASM is not found on path.
