--- kvazaar-1.1.0/configure.ac.orig	2017-01-31 15:28:02.000000000 +0100
+++ kvazaar-1.1.0/configure.ac	2017-06-06 22:24:54.528272790 +0200
@@ -95,9 +95,10 @@
 
 
 # host and cpu specific settings
-AS_CASE([$host_cpu],
-        [i?86], [BITS="32" ASFLAGS="$ASFLAGS -DARCH_X86_64=0" X86="true"],
-        [x86_64], [BITS="64" ASFLAGS="$ASFLAGS -DARCH_X86_64=1 -m amd64" X86="true"],
+AS_CASE([$host],
+        [i?86-*], [BITS="32" ASFLAGS="$ASFLAGS -DARCH_X86_64=0" X86="true"],
+        [x86_64-*-gnux32|amd64-*-gnux32], [BITS="x32" ASFLAGS="$ASFLAGS -DARCH_X86_64=1 -m x32" X86="true"],
+        [x86_64-*|amd64-*], [BITS="64" ASFLAGS="$ASFLAGS -DARCH_X86_64=1 -m amd64" X86="true"],
         [powerpc*], [PPC="true"]
 )
 
